[Lam Rong Yi] iP

lrongyi opened the pull request:
# **DOOK**

Dook is an all in one chatbot that helps you to keep track of all your tasks at hand! 💯 💥 💫 💨 

Dook is better than other chatbots, it

- _is text-based_
- ~~always replies~~ 
- **easy to use**

> "I just can't stop using Dook" 

The commands Dook supports are: 

- [x] Marking/Unmarking Tasks as Done
- [x] Creating/Deleting Tasks
- [x] Listing all Tasks
- [x] Searching for Tasks

How to get Dook: 

1. Check it out [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html)
2. Add your tasks
3. Let Dook know when you're done

Here's a `main` method: 

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

lrongyi committed:
Rename, Greet, Exit

lrongyi committed:
Rename, Greet, Exit

lrongyi committed:
Echo

lrongyi committed:
Add, List

lrongyi committed:
Mark as Done

lrongyi committed:
ToDos, Events, Deadlines

lrongyi committed:
Automated Text UI Testing

lrongyi committed:
Handle Errors

lrongyi committed:
Delete

lrongyi committed:
Update Dook.java

lrongyi committed:
Implement save function

lrongyi committed:
Merge branch 'Level-7'

lrongyi committed:
Merge branch 'master' of https://github.com/lrongyi/ip

lrongyi committed:
Add DateTime format to dates

lrongyi committed:
Merge branch 'branch-Level-8'

lrongyi committed:
Make the code more OOP

lrongyi committed:
Divide classes into packages

lrongyi committed:
Merge branch 'A-Packages'

lrongyi committed:
Change README

lrongyi committed:
Merge branch 'add-gradle-support'

lrongyi committed:
Use Gradle and Add JUnit Tests

lrongyi committed:
Update README.md

lrongyi committed:
Implement Find command

lrongyi committed:
Use Gradle and Add JUnit Tests

lrongyi committed:
Test commit

lrongyi committed:
Follow coding standards

lrongyi committed:
Add JavaDoc comments

lrongyi committed:
Merge branch 'branch-Level-9'

lrongyi committed:
Merging branch 'branch-A-CodingStandard'

lrongyi committed:
Merging 'branch-A-JavaDoc'

lrongyi committed:
Add Checkstyle and format code

lrongyi committed:
Merge branch 'A-Checkstyle'

yxtan02 commented on src/main/java/dook/Dook.java:
```diff
@@ -0,0 +1,47 @@
+import dook.tasks.TaskList;
+import dook.ui.Ui;
+
+public class Dook {
```
Would it be better to add Javadoc header comment for this class? According to the coding standards, header comments should be written for all public classes/methods.

yxtan02 commented on src/main/java/dook/commands/CreateCommand.java:
```diff
@@ -0,0 +1,110 @@
+        try {
+            Task task;
+
+            switch (this.type) {
+            case TODO:
+                task = new Todo(this.description);
+                break;
+            case DEADLINE:
+                LocalDateTime time = LocalDateTime.parse(this.by, formatter);
+                task = new Deadline(this.description, time);
+                break;
+            case EVENT:
+                LocalDateTime startTime = LocalDateTime.parse(this.start, formatter);
+                LocalDateTime endTime = LocalDateTime.parse(this.end, formatter);
+                task = new Event(this.description, startTime, endTime);
+                break;
+            default:
+                throw new DookException("Invalid task type");
```
Good job for following the coding standards for switch statement (no indentation for case clauses) !

yxtan02 commented on src/main/java/dook/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,36 @@
+    }
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DookException, IOException {
+        Task removed = tasks.delete(taskNumber - 1);
```
Would it be better to rename the variable removed to removed task since variable names should be nouns?

yxtan02 commented on src/main/java/dook/commands/TaskType.java:
```diff
@@ -0,0 +1,5 @@
+package dook.commands;
+
+public enum TaskType {
+    TODO, EVENT, DEADLINE
+};
```
Should this class be moved to the task package since the enum TaskType is related to tasks?

yxtan02 commented on src/test/java/dook/commands/Command.java:
```diff
@@ -0,0 +1,19 @@
+import dook.ui.Ui;
+
+
+public abstract class Command {
```
Good use of abstract class to extract out common attributes and methods for various command classes

yxtan02 reviewed (commented):
Looks good to me. Just a few nits to fix.

lrongyi committed:
Add a GUI to the chatbot

lrongyi committed:
Merge branch 'branch-Level-10'

rahula1008 commented on src/main/java/dook/Dook.java:
```diff
@@ -0,0 +1,71 @@
+import dook.tasks.TaskList;
+import dook.ui.Ui;
+
+public class Dook {
```
Could it be a good idea to add JavaDoc comments as this is a non-trivial public class?

rahula1008 commented on src/main/java/dook/Dook.java:
```diff
@@ -0,0 +1,71 @@
+        }
+    }
+
+    public String greet() {
```
Any reason why you added another function for greet instead of calling the original function?

rahula1008 commented on src/main/java/dook/Dook.java:
```diff
@@ -0,0 +1,71 @@
+import dook.parser.Parser;
+import dook.storage.Storage;
+import dook.tasks.TaskList;
+import dook.ui.Ui;
```
I like how you ordered your import statements within each block alphabetically.

rahula1008 commented on src/main/java/dook/commands/Command.java:
```diff
@@ -0,0 +1,18 @@
+import dook.tasks.TaskList;
+import dook.ui.Ui;
+
+public abstract class Command {
```
Could it be a good idea to use Javadoc header comments as it is a non-trivial public class?

rahula1008 commented on src/main/java/dook/commands/CreateCommand.java:
```diff
@@ -0,0 +1,113 @@
+package dook.commands;
```
I like how you organised your code into well-organised and well-named packages.

rahula1008 commented on src/main/java/dook/commands/TaskType.java:
```diff
@@ -0,0 +1,5 @@
+package dook.commands;
+
+public enum TaskType {
```
Any reason why you put this in the commands package as opposed to the tasks package?

rahula1008 commented on src/main/java/dook/parser/Parser.java:
```diff
@@ -0,0 +1,59 @@
+            String command = components[0];
+
+            switch (command) {
+            case "bye":
```
I like how you adhered to the coding standard guideline on switch statements.

rahula1008 commented on src/main/java/dook/tasks/Task.java:
```diff
@@ -0,0 +1,37 @@
+        this.isDone = false;
+    }
+
+    public String getDescription() {
```
I like that even though these are public methods, since they are getters / setters, you omitted the JavaDoc, as it says in the coding standard that you may omit Javadocs for getters and setters.

rahula1008 commented on src/main/java/dook/tasks/TaskList.java:
```diff
@@ -0,0 +1,49 @@
+        return removed;
+    }
+
+    public Task add(Task task) {
```
Could it be a good idea to add some JavaDocs here just to remind the user that this method also returns the task?

rahula1008 reviewed (commented):
Overall, I liked how you structured your code into well-named packages and your code is easy to read. I noted a few minor coding standard violations
too. Some of the classes are missing some Javadocs, so consider adding some more Javadocs if that makes sense.

lrongyi committed:
Add missing Javadocs to all the Commands

- Add Javadocs to all Commands, detailing the functionality of their
respective 'execute' methods
- Add Javadocs to TaskList 'delete' method to specify that it returns a
list
- Refactor TaskType enum to be in the Task package instead of the
Command package

This improves code readability and facilitates easier maintenance.

lrongyi committed:
Add assertions to improve correctness and reliability of code

- Add asserions to the TaskList class to validate index and null values
- Add assertions to FindCommand and MarkingCommands to validate null
  values
- Add Javadocs for Dook and DookException class

These will help to improve code readability and reliability.

lrongyi committed:
Merge pull request #1 from lrongyi/branch-A-Assertions

Add assertions to improve correctness and reliability of code

lrongyi committed:
Improve code quality

lrongyi committed:
Merge branch 'master' into branch-A-CodeQuality

lrongyi committed:
Merge pull request #2 from lrongyi/branch-A-CodeQuality

Improve code quality

lrongyi committed:
Bug fix

lrongyi committed:
Add user guide using the new Help Command

lrongyi committed:
Merge pull request #3 from lrongyi/branch-BCD-Extension

Add user guide using the new Help Command

lrongyi committed:
Update JavaDocs

lrongyi committed:
Resolve weak SLAP

lrongyi committed:
Update README.md

lrongyi committed:
Add emojis to README.md

lrongyi committed:
Update GUI and Ui.png

lrongyi committed:
Create more JUnit Tests
